databuilder/extractor/dashboard/tableau/tableau_dashboard_extractor.py

```diff
@@ -0,0 +1,139 @@
+import logging
+
+from pyhocon import ConfigTree, ConfigFactory  # noqa: F401
+from typing import Any  # noqa: F401
+
+from databuilder import Scoped
+
+from databuilder.extractor.base_extractor import Extractor
+from databuilder.extractor.restapi.rest_api_extractor import STATIC_RECORD_DICT
+
+import databuilder.extractor.dashboard.tableau.tableau_dashboard_constants as const
+from databuilder.extractor.dashboard.tableau.tableau_dashboard_utils import TableauDashboardAuth,\
+    TableauGraphQLApiExtractor, TableauDashboardUtils
+
+from databuilder.transformer.base_transformer import ChainedTransformer
+from databuilder.transformer.dict_to_model import DictToModel, MODEL_CLASS
+from databuilder.transformer.timestamp_string_to_epoch import TimestampStringToEpoch, FIELD_NAME
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TableauDashboardExtractor(Extractor):
+    """
+    Extracts core metadata about Tableau "dashboards".
+    For the purposes of this extractor, Tableau "workbooks" are mapped to Amundsen dashboards, and the
+    top-level project in which these workbooks preside is the dashboard group. The metadata it gathers is:
+        Dashboard name (Workbook name)
+        Dashboard description (Workbook description)
+        Dashboard creation timestamp (Workbook creationstamp)
+        Dashboard group name (Workbook top-level folder name)
+    As with all the Tableau extractors, uses the Metadata API: https://help.tableau.com/current/api/metadata_api/en-us/index.html
+    """
+
+    API_VERSION = const.API_VERSION
+    TABLEAU_HOST = const.TABLEAU_HOST
+    SITE_NAME = const.SITE_NAME
+    TABLEAU_ACCESS_TOKEN_NAME = const.TABLEAU_ACCESS_TOKEN_NAME
+    TABLEAU_ACCESS_TOKEN_SECRET = const.TABLEAU_ACCESS_TOKEN_SECRET
+    EXCLUDED_PROJECTS = const.EXCLUDED_PROJECTS
+    EXTERNAL_CLUSTER_NAME = const.EXTERNAL_CLUSTER_NAME
+    EXTERNAL_SCHEMA_NAME = const.EXTERNAL_SCHEMA_NAME
+    EXTERNAL_TABLE_TYPES = const.EXTERNAL_TABLE_TYPES
```

Comment:
I do like how these configs are now more visible as an interface. But, we should limit them to only those that are used in this extractor so these `EXTERNAL_*` ones should be omitted. This can be applied to all entry extractor classes and the downstream graphql ones too.